Source/Scene/QuadtreePrimitive.js

```diff
@@ -461,7 +513,13 @@ define([
         tilesToRender.sort(tileDistanceSortFunction);
 
         for (var i = 0, len = tilesToRender.length; i < len; ++i) {
-            tileProvider.showTileThisFrame(tilesToRender[i], context, frameState, commandList);
+            var tile = tilesToRender[i];
+            tileProvider.showTileThisFrame(tile, context, frameState, commandList);
+
+            if (tile._frameRendered !== frameState.frameNumber - 1) {
+                primitive.tileVisibleEvent.raiseEvent(tile);
```

Comment:
Perhaps we can introduce the intermediate per-position notification layer and time slice by queueing up these events and servicing them after tree traversal.